shopping-cart/shopping-cart-scala/shopping-cart/src/main/scala/com/example/shoppingcart/impl/ShoppingCart.scala

```diff
@@ -0,0 +1,250 @@
+import akka.cluster.sharding.typed.scaladsl._
+import akka.persistence.typed.PersistenceId
+import akka.persistence.typed.scaladsl.RetentionCriteria
+import akka.persistence.typed.scaladsl.Effect
+import akka.persistence.typed.scaladsl.EventSourcedBehavior
+import akka.persistence.typed.scaladsl.ReplyEffect
+import com.lightbend.lagom.scaladsl.persistence.AggregateEvent
+import com.lightbend.lagom.scaladsl.persistence.AggregateEventShards
+import com.lightbend.lagom.scaladsl.persistence.AggregateEventTag
+import com.lightbend.lagom.scaladsl.persistence.AggregateEventTagger
+import com.lightbend.lagom.scaladsl.persistence.AkkaTaggerAdapter
+import com.lightbend.lagom.scaladsl.playjson.JsonSerializer
+import com.lightbend.lagom.scaladsl.playjson.JsonSerializerRegistry
+import play.api.libs.json.Format
+import play.api.libs.json._
+
+import scala.collection.immutable.Seq
+
+object ShoppingCart {
+
+  // SHOPPING CART COMMANDS
+
+  // This is a marker trait for shopping cart commands.
+  // We will serialize them using Akka's Jackson support that is able to deal with the replyTo field.
+  // (see application.conf).
+  // Keep in mind that when configuring it on application.conf you need to use the FQCN which is:
+  // com.example.shoppingcart.impl.ShoppingCart$CommandSerializable
+  // Note the "$".
+  trait CommandSerializable
+
+  sealed trait Command extends CommandSerializable
+
+  final case class AddItem(itemId: String, quantity: Int, replyTo: ActorRef[Confirmation]) extends Command
+
+  final case class RemoveItem(itemId: String, replyTo: ActorRef[Confirmation]) extends Command
+
+  final case class AdjustItemQuantity(itemId: String, quantity: Int, replyTo: ActorRef[Confirmation]) extends Command
+
+  final case class Checkout(replyTo: ActorRef[Confirmation]) extends Command
+
+  final case class Get(replyTo: ActorRef[Summary]) extends Command
+
+  // SHOPPING CART EVENTS
+  sealed trait Event extends AggregateEvent[Event] {
+    override def aggregateTag: AggregateEventTagger[Event] = Event.Tag
+  }
+
+  object Event {
+    val Tag: AggregateEventShards[Event] = AggregateEventTag.sharded[Event](numShards = 10)
+  }
+
+  final case class ItemAdded(itemId: String, quantity: Int) extends Event
+
+  final case class ItemRemoved(itemId: String) extends Event
+
+  final case class ItemQuantityAdjusted(itemId: String, newQuantity: Int) extends Event
+
+  final case class CartCheckedOut(eventTime: Instant) extends Event
+
+  // Events get stored and loaded from the database, hence a JSON format
+  //  needs to be declared so that they can be serialized and deserialized.
+  implicit val itemAddedFormat: Format[ItemAdded]                       = Json.format
+  implicit val itemRemovedFormat: Format[ItemRemoved]                   = Json.format
+  implicit val itemQuantityAdjustedFormat: Format[ItemQuantityAdjusted] = Json.format
+  implicit val cartCheckedOutFormat: Format[CartCheckedOut]             = Json.format
+
+  // SHOPPING CART REPLIES
```

Comment:
Minor request that will make the creation of snippets in docs simpler. Keep Replies immediately after the commands. In https://github.com/lagom/lagom/pull/2388 the replies and commands are introduced and detailed as a single section.